tensorzero-core/src/inference/types/file.rs

```diff
@@ -167,6 +140,72 @@ impl File {
                     })
                 })?;
 
+                let mime_type = if let Some(mime_type) = mime_type {
+                    // Priority 1: Explicitly provided mime_type
+                    mime_type
+                } else {
+                    // Priority 2: Infer from file content
+                    let inferred = infer::get(&bytes[..std::cmp::min(8192, bytes.len())]);
+
+                    if let Some(inferred_type) = inferred {
+                        let inferred_mime = inferred_type
+                            .mime_type()
+                            .parse::<MediaType>()
+                            .map_err(|e| {
+                                Error::new(ErrorDetails::BadImageFetch {
+                                    url: url.clone(),
+                                    message: format!("Inferred mime type is not valid: {e}"),
+                                })
+                            })?;
+
+                        // Check if Content-Type header differs and log warning
+                        if let Some(content_type) = &content_type_header {
+                            if let Ok(content_type_str) = content_type.to_str() {
+                                if let Ok(header_mime) = content_type_str.parse::<MediaType>() {
+                                    if header_mime != inferred_mime {
+                                        tracing::warn!(
+                                            "Inferred MIME type '{}' differs from Content-Type header '{}' for URL {}",
+                                            inferred_mime,
+                                            header_mime,
+                                            url
```

Comment:
Clippy is going to complain about this.

Also append to the message something:

```
... The gateway will send `{}` to the model provider.
```

Also use ` instead of ' for proper formatting in general.